x-pack/plugins/endpoint/public/applications/endpoint/lib/location/is_on_page.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EndpointAppLocation } from '../../types';
+
+let currentLocation: EndpointAppLocation;
+let fromLocation: EndpointAppLocation;
+
+export const storeCurrentLocation = (loc: EndpointAppLocation) => {
+  fromLocation = currentLocation;
+  currentLocation = loc;
+};
+
+export const isOnPage = (pageRoute: EndpointAppLocation['pathname']) => {
```

Comment:
I'm thinking `pageRoute` should really be `routePath` and we would used `reat-router` `match` object to determine if on a given path. Example: `isOnPage('/policy/:id')`. Internally, the check would be something like `currentLocation.match.path === routePath`

See my comment below about using `match` for more on this.

I would propose that this lib only contain a `isOnPage()` and a `wasOnPage()` method.  We can then create sub-libs (as you did for policy) that have domain specific methods (e.x. `isOnPolicyListPage()`, isOnPolicyDetailsPage()`, etc...)